pyntc/devices/ios_device.py

```diff
@@ -541,12 +543,53 @@ def file_copy_remote_exists(self, src, dest=None, file_system=None):
             return True
         return False
 
-    def install_os(self, image_name, **vendor_specifics):
+    def install_os(self, image_name, install_mode=False, **vendor_specifics):
+        """Installs the prescribed Network OS, which must be present before issuing this command.
+
+        Args:
+            image_name (str): Name of the IOS image to boot into
+            install_mode (bool, optional): Uses newer install method on devices. Defaults to False.
+
+        Raises:
+            OSInstallError: Unable to install OS Error type
+
+        Returns:
+            bool: False if no install is needed, true if the install completes successfully
+        """
         timeout = vendor_specifics.get("timeout", 3600)
         if not self._image_booted(image_name):
-            self.set_boot_options(image_name, **vendor_specifics)
-            self.reboot(confirm=True)
+            if install_mode:
+                # Change boot statement to be boot system <flash>:packages.conf
+                self.set_boot_options(INSTALL_MODE_FILE_NAME, **vendor_specifics)
+                delay_factor = 10
+
+                # Check for OS Version specific upgrade path
+                # https://www.cisco.com/c/en/us/td/docs/switches/lan/catalyst9300/software/release/17-2/release_notes/ol-17-2-9300.html
+                os_version = self.os_version
+                if "16.5.1a" in os_version or "16.6.1" in os_version:
```

Comment:
Do we want to hard code OS versions? What if they change / become obsolete? Maybe add a constant at the beginning that can be changed?

Reply:
These are actually quite old versions at this point that we have moved on from. Current versions are 17.1 & 17.2, with a whole bunch of other dot versions 16.5, 16.6, 16.7, 16.8, 9, and more in between. I was thinking originally not including this at all, but it makes sense to cover our bases there.

Do you have any recommended methods of how to search list items in a string? The string inside of a list is easy. This one I'm not sure on.

Reply:
If we want to stick on this but the tests are all passing, can we add an issue to update later?

Reply:
I am not sure if this is the most efficient way but I think you could make a list of os_versions then maybe check if the current os version is in this list:
```
list_os_versions = ['17.1', '17.2', '16.5', '16.7', '16.8', '16.9']
if os_version in list_os_versions:
    ...
```
Please let me know if I did not understand what you were trying to achieve.

Reply:
@mundruid I started to suggest something similar to this yesterday, but the lookup needs to happen in revers order, since `os_version` will be more than just the "16.x" portion. Whatever the final state of this PR is, we will want to go back later and refactor to handle install vs bundle modes better

Reply:
Got it @jmcgill298 , thanks for clarifying. Then it looks good to me as well!